feature-libs/quote/components/links/quote-links.component.spec.ts

```diff
@@ -168,4 +208,72 @@ describe('QuoteLinksComponent', () => {
 
     expect(router.url).toBe('/cxRoute:quotes');
   }));
+
+  describe('Download proposal document', () => {
+    const vendorQuote: Quote = {
+      ...mockQuote,
+      sapAttachments: [
+        {
+          id: mockQuote.code,
+        },
+      ],
+    };
+
+    it('should display download button if there is a proposal document attached to the quote', () => {
+      mockQuoteDetails$.next(vendorQuote);
+      fixture.detectChanges();
+      const buttonContainerSection = CommonQuoteTestUtilsService.getHTMLElement(
+        htmlElem,
+        'section'
+      );
+      CommonQuoteTestUtilsService.expectElementPresent(
+        expect,
+        buttonContainerSection,
+        'button'
+      );
+      CommonQuoteTestUtilsService.expectElementToContainText(
+        expect,
+        htmlElem,
+        'button',
+        'download'
+      );
+    });
+
+    it('should not display download button if there is no proposal document attached to the quote', () => {
+      mockQuoteDetails$.next(mockQuote);
+      fixture.detectChanges();
+      const buttonContainerSection = CommonQuoteTestUtilsService.getHTMLElement(
+        htmlElem,
+        'section'
+      );
+      CommonQuoteTestUtilsService.expectElementNotPresent(
+        expect,
+        buttonContainerSection,
+        'button'
+      );
+    });
+
+    it('should download the proposal document attached when Download button is clicked', () => {
+      spyOn(quoteFacade, 'downloadAttachment').and.returnValue(
+        of(mockQuoteAttachment())
+      );
+      spyOn(fileDownloadService, 'download');
+      mockQuoteDetails$.next(vendorQuote);
+      fixture.detectChanges();
+      const downloadBtn = CommonQuoteTestUtilsService.getHTMLElement(
+        htmlElem,
+        'button'
+      );
+      downloadBtn.click();
+      fixture.detectChanges();
+      expect(quoteFacade.downloadAttachment).toHaveBeenCalledWith(
+        vendorQuote.code,
+        vendorQuote.code
+      );
+      fixture.whenStable().then(() => {
+        fixture.detectChanges();
+        expect(fileDownloadService.download).toHaveBeenCalled();
```

Comment:
This assertion does not work. See test output:

```
ERROR: Error: <toHaveBeenCalled> : Expected a spy, but got Function.
Usage: expect(<spyObj>).toHaveBeenCalled()
Error: <toHaveBeenCalled> : Expected a spy, but got Function.
Usage: expect(<spyObj>).toHaveBeenCalled()
    at compare (http://localhost:9876/absolute/Users/D026976/spa/spartacus/node_modules/jasmine-core/lib/jasmine-core/jasmine.js?4d3d165a31f4a0cfbea94aa42ec76e0330a3121f:6023:17)
```

Reply:
Fixed the error.